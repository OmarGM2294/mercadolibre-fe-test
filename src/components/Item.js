import React from 'react'
import Image from 'next/image'

import { useRouter } from 'next/router'

import styles from '../../styles/components/Item.module.scss'

const Item = (props) => {
  const router = useRouter()

  return (
    <article className={styles.item} onClick={() => router.push(`/items/${props.data.id}`)}>
      <div className={styles.container}>
        <Image
          className={styles.image}
          src={props.data.picture}
          alt={props.data.title}
          width={180}
          height={180}
        />
        <div className={styles.info}>
          <div className={styles.status}>
            <div className={styles.values}>
              <h2>
                $ {props.data.price.amount}
              </h2>
              { props.data.free_shipping ?
                <Image
                  className={styles.shipping}
                  src="/imgs/shipping.png"
                  alt="Shipping"
                  height={20}
                  width={20}
                />
              : null }
            </div>
            <span>{props.data.state_name}</span>
          </div>
          <p>{props.data.title}</p>
        </div>
      </div>
    </article>
  )
}

export default Item