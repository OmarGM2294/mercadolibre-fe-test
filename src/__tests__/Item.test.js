import React from 'react'

import { create } from 'react-test-renderer'

import Item from '../components/Item'

describe('Item', () => {
  it('Should match snapshot (it renders correctly)', () => {
    const component = create(<Item data={{ 
      picture: '/imgs/search.svg',
      title: '',
      price: {
        amount: 0,
      },
      state_name: ''
    }}/>)

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
