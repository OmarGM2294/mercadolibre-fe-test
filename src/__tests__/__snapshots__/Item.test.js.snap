// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Should match snapshot (it renders correctly) 1`] = `
<article>
  <div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgwIiBoZWlnaHQ9IjE4MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt=""
          decoding="async"
          src="/imgs/search.svg"
          srcSet="/_next/image?url=%2Fimgs%2Fsearch.svg&w=256&q=75 1x, /_next/image?url=%2Fimgs%2Fsearch.svg&w=384&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt=""
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
    <div>
      <div>
        <h2>
          $ 
          0
        </h2>
        <span>
          
        </span>
      </div>
      <p>
        
      </p>
    </div>
  </div>
</article>
`;
